src/utils/tests/cache.rs

```diff
@@ -0,0 +1,49 @@
+use crate::utils::cache::CachedWithCode;
```

Comment:
We are missinga tests with multiple seperators: `404;foo;bar`. For that I would want to have
```
CachedWithCode {
  code: 404,
  data: String::from("foo;bar"),
};
```